RetroAchievements/RAWeb
# perf(modal-trigger): migrate alpine component to app bundle

This PR resolves a performance issue in the `<x-modal-trigger>` component. If there are multiple triggers on the page, the `<script>` tag in the Blade component is being embedded multiple times, resulting in a bloated DOM size.

To mitigate this issue, the Alpine.js code has been moved to the app bundle.

**How to test this PR:**
* Go to a game page and click on a type indicator (while logged in) to see a lazy-loaded dialog appear.
* Go to reportissue.php and click on one of the helper buttons to see one of the static dialogs appear.
